internal/internal_task_pollers.go

```diff
@@ -1148,7 +1148,7 @@ func (atp *activityTaskPoller) ProcessTask(task interface{}) error {
 	}
 
 	activityTask := task.(*activityTask)
-	if activityTask.task == nil {
+	if activityTask.task == nil || len(activityTask.task.TaskToken) == 0 {
```

Comment:
do we need a similar change on workflow tasks?  since that path was changed too.

Reply:
WorkflowTask doesn't need change since it's already handling emptiness [here](https://github.com/shijiesheng/cadence-client/blob/c06b7bb191cd4bdc9b57307607e3f7b40855bc1d/internal/internal_task_handlers.go#L751-L756)

Reply:
👍 thanks, I'm convinced :)  same kinds of paths as before then, should be safe.